presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestExpireSnapshotProcedure.java

```diff
@@ -0,0 +1,289 @@
+        table.refresh();
+        assertHasSize(table.snapshots(), 0);
+
+        assertUpdate(format("CALL system.expire_snapshots(schema => '%s', table_name => '%s')", TEST_SCHEMA, tableName));
+        assertUpdate(format("CALL system.expire_snapshots(schema => '%s', table_name => '%s', older_than => %s)", TEST_SCHEMA, tableName, "timestamp '1984-12-08 00:10:00.000'"));
+        assertUpdate(format("CALL system.expire_snapshots(schema => '%s', table_name => '%s', retain_last => %s)", TEST_SCHEMA, tableName, 5));
+        assertUpdate(format("CALL system.expire_snapshots(schema => '%s', table_name => '%s', snapshot_ids => %s)", TEST_SCHEMA, tableName, "ARRAY[12345]"));
+        table.refresh();
+        assertHasSize(table.snapshots(), 0);
+
+        dropTable(tableName);
+    }
+
+    @Test(dataProvider = "timezones")
+    public void testExpireSnapshotsUsingPositionalArgs(String zoneId, boolean legacyTimestamp)
+    {
+        Session session = sessionForTimezone(zoneId, legacyTimestamp);
+        String tableName = "positional_args_table";
+        try {
+            assertUpdate(session, "CREATE TABLE " + tableName + " (id integer, value varchar)");
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(1, 'a')", 1);
+
+            Table table = loadTable(tableName);
+            Snapshot firstSnapshot = table.currentSnapshot();
+
+            waitUntilAfter(firstSnapshot.timestampMillis());
+
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(2, 'b')", 1);
+            table.refresh();
+
+            Snapshot secondSnapshot = table.currentSnapshot();
+            String formattedDateTime = getTimestampString(secondSnapshot.timestampMillis(), zoneId);
+
+            assertHasSize(table.snapshots(), 2);
+
+            // expire without retainLast param
+            String callStr = format("CALL system.expire_snapshots('%s', '%s', TIMESTAMP '%s')", TEST_SCHEMA, tableName, formattedDateTime);
+            assertUpdate(session, callStr);
+
+            table.refresh();
+            assertHasSize(table.snapshots(), 1);
+
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(3, 'c')", 1);
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(4, 'd')", 1);
+            assertQuery(session, "select * from " + tableName, "values(1, 'a'), (2, 'b'), (3, 'c'), (4, 'd')");
+
+            table.refresh();
+
+            waitUntilAfter(table.currentSnapshot().timestampMillis());
+
+            String currentTimestamp = getTimestampString(System.currentTimeMillis(), zoneId);
+
+            assertHasSize(table.snapshots(), 3);
+
+            // expire with retainLast param
+            assertUpdate(session, format("CALL system.expire_snapshots('%s', '%s', TIMESTAMP '%s')", TEST_SCHEMA, tableName, currentTimestamp));
```

Comment:
Am I missing something? It seems like the `retainLast` param isn't specified here even though the comment indicates you should be using it.

Reply:
Thanks for the pointer. My fault for losing the retainLast parameter, fixed!